src/fsstorage.cc

```diff
@@ -13,35 +13,32 @@ FSStorage::FSStorage(const Config& config) : config_(config) {
 }
 FSStorage::~FSStorage() {
   // TODO: clear director_files, image_files
+  clearPrimaryKeys();
 }
 
 void FSStorage::storePrimaryKeys(const std::string& public_key, const std::string& private_key) {
-  boost::filesystem::path public_key_path = config_.tls.certificates_directory / config_.uptane.public_key_path;
-  boost::filesystem::path private_key_path = config_.tls.certificates_directory / config_.uptane.private_key_path;
+  boost::filesystem::remove(config_.uptane.public_key_path);
```

Comment:
Regarding the names see the discussion in https://github.com/advancedtelematic/aktualizr/pull/324